src/lib.rs

```diff
@@ -228,10 +228,10 @@ pub enum Expression {
 }
 
 pub type Block = Vec<Statement>;
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub struct FallThrough;
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
```

Comment:
It seems to me that we could consistently return handles to things in the error instead of their copies. If true, that would allow us to not derive `Clone` for these types, while the users would still be able to get the actual data from the module arenas. Would you be open to explore this direction?

Reply:
I just took a look at what this would involve. I think it would require a decent amount of refactoring, because we'd need to change a lot of functions from taking `&X` to `Handle<X>`.

Just for my own understanding, how bad is it for these types to implement `Clone`? My reasoning was that cloning the types to return errors would be fine because it wouldn't slow down the hot path.

Reply:
Yeah, it's not a concern about performance. Rather, if this is meant to only be used by error reporting, I don't want users of Naga to consider this API to be public, unless there is a strong case for it.
Given your findings, we can go ahead with this. Let's please add `//Note` comments above these structs saying that `Clone` is only used by the error reporting.

Reply:
Is the following note okay?

```
/// Clone is used only for error reporting and is not intended for end users
```

Should I add this note for any data struct that implements `Clone`? Or just stuff that I am adding in this PR? E.g. `ScalarKind` previously derived `Clone`.

Reply:
I wasn't considering to make this a doc comment. Rather - a note for us internally.
It only needs to be on the stuff that you added.

Reply:
All right, I added this note for each new derived clone.